### PR TITLE
Fix some issues found by LGTM.com

### PR DESCRIPTION
**Pull request by AlCalzone**

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -1214,12 +1214,6 @@ function Adapter(options) {
                 return;
             }
 
-            if (!obj) {
```
Comment (AlCalzone) [inline lib/adapter.js]: This is checked in Line 1206

File: lib/adapter.js [hunk 2]
```diff
@@ -1525,8 +1519,9 @@ function Adapter(options) {
             }
 
             // delete arrays if they should be changed
-            if (obj && ((obj.native && (obj.native.repositories || obj.native.certificates || obj.native.devices)) ||
```
Comment (AlCalzone) [inline lib/adapter.js]: Existence of obj is checked in Line 1516

File: lib/adapter.js [hunk 3]
```diff
@@ -1557,7 +1552,7 @@ function Adapter(options) {
 
                     this.objects.setObject(id, obj, callback);
                 });
-            } else if (obj) {
```
Comment (AlCalzone) [inline lib/adapter.js]: dito

File: lib/cli/cliLogs.js [hunk 4]
```diff
@@ -1,11 +1,10 @@
 'use strict';
 // const CLI = require('./messages.js');
 const CLICommand = require('./cliCommand.js');
-const { getDefaultDataDir, getConfigFileName } = require('../tools');
+const { getConfigFileName } = require('../tools');
 const chokidar = require('chokidar');
 const fs = require('fs');
 const os = require('os');
-const path = require('path');
```
Comment (AlCalzone) [inline lib/cli/cliLogs.js]: Both imports unused

File: lib/cli/cliProcess.js [hunk 5]
```diff
@@ -163,8 +163,8 @@ module.exports = class CLIProcess extends CLICommand {
                 if (fs.existsSync(killAllScriptPath)) {
                     fs.chmodSync(killAllScriptPath, '777');
                     const child = require('child_process').spawn(killAllScriptPath, []);
-                    child.stdout.on('data', _data => data += _data.toString().replace('\n', ''));
-                    child.stderr.on('data', _data => data += _data.toString().replace('\n', ''));
+                    child.stdout.on('data', _data => data += _data.toString().replace(/\n/g, ''));
+                    child.stderr.on('data', _data => data += _data.toString().replace(/\n/g, ''));
```
Comment (AlCalzone) [inline lib/cli/cliProcess.js]: We should replace all occurences, not only the first

File: lib/logger.js [hunk 6]
```diff
@@ -70,7 +70,7 @@ const logger = function (level, files, noStdout, prefix) {
             winstonFormats.push(winston.format.json());
         }
         if (prefix) {
-            winstonFormats.push(winston.format.label({ label: prefix || '' }));
+            winstonFormats.push(winston.format.label({ label: prefix }));
```
Comment (AlCalzone) [inline lib/logger.js]: prefix is guaranteed to exist (Line 72)

File: lib/objects/objectsInMemServer.js [hunk 7]
```diff
@@ -1880,7 +1880,7 @@ function ObjectsInMemServer(settings) {
 
         obj._id = id;
 
-        if (id === 'system.config' && obj && obj.common && objects[id] && objects[id].common && JSON.stringify(obj.common.defaultNewAcl) !== JSON.stringify(objects[id].common.defaultNewAcl)) {
+        if (id === 'system.config' && obj.common && objects[id] && objects[id].common && JSON.stringify(obj.common.defaultNewAcl) !== JSON.stringify(objects[id].common.defaultNewAcl)) {
```
Comment (AlCalzone) [inline lib/objects/objectsInMemServer.js]: obj is guaranteed to exist (Line 1873)

File: lib/objects/objectsUtils.js [hunk 8]
```diff
@@ -462,13 +462,10 @@ function sanitizePath(id, name, callback) {
         return;
     }
 
-    if (id) {
-        id = id.replace(/\.\./g, ''); // do not allow to write in parent directories
-    }
+    id = id.replace(/\.\./g, ''); // do not allow to write in parent directories
```
Comment (AlCalzone) [inline lib/objects/objectsUtils.js]: ID is guaranteed to exist (Line 458)

File: lib/objects/objectsUtils.js [hunk 9]
```diff
@@ -462,13 +462,10 @@ function sanitizePath(id, name, callback) {
...
     }
 
-    if (id) {
-        id = id.replace(/\.\./g, ''); // do not allow to write in parent directories
-    }
+    id = id.replace(/\.\./g, ''); // do not allow to write in parent directories
 
     if (name.indexOf('..') !== -1) {
         name = path.normalize(name);
-        name = name.replace(/\//g, '/');
```
Comment (AlCalzone) [inline lib/objects/objectsUtils.js]: This replaces `/` with itself

File: lib/states/statesInRedis.js [hunk 10]
```diff
@@ -124,7 +124,11 @@ function StateRedis(settings) {
      * @param callback {Function}   will be called when redis confirmed reception of the command
      */
     this.setState = function (id, state, callback) {
-        if (!client) return;
+
+        if (!client) {
+            return callback && callback('Closed');
+        }
+
```
Comment (AlCalzone) [inline lib/states/statesInRedis.js]: Duplicate check. I assume you want to keep the one which invokes the callback.

File: lib/adapter.js (hunk 1 above)
Comment (GermanBluefox) [reply lib/adapter.js]: Ok

**Pull request merged**